packages/react/src/components/Button/Button.tsx

```diff
@@ -84,120 +97,172 @@ export interface ButtonProps
+  const hasChildren = childrenExist(children)
+
+  setStart()
+
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: Button.displayName,
+    mapPropsToBehavior: () => ({
+      as,
+      active,
+      disabled,
+      loading,
+    }),
+    actionHandlers: {
+      performClick: event => {
+        event.preventDefault()
+        handleClick(event)
+      },
+    },
+    rtl: context.rtl,
+  })
+  const { classes, styles: resolvedStyles } = useStyles(Button.displayName, {
+    className: Button.className,
+    mapPropsToStyles: () => ({
+      text,
+      primary,
+      disabled,
+      circular,
+      size,
+      loading,
+      inverted,
+      iconOnly,
+      fluid,
+      hasContent: !!content,
+    }),
+    mapPropsToInlineStyles: () => ({
+      unstable_animation: animation,
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  })
 
-  renderIcon = (variables, styles) => {
-    const { icon, iconPosition, content } = this.props
+  const unhandledProps = getUnhandledProps(Button.handledProps, props)
+  const ElementType = getElementType(props)
 
+  const renderIcon = () => {
     return Icon.create(icon, {
-      defaultProps: () => ({
-        styles: styles.icon,
-        xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
-        variables: variables.icon,
-      }),
+      defaultProps: () =>
+        getA11Props('icon', {
+          styles: resolvedStyles.icon,
+          xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
+        }),
     })
   }
 
-  renderLoader = (variables, styles) => {
-    const { loader } = this.props
-
+  const renderLoader = () => {
     return Loader.create(loader || {}, {
-      defaultProps: () => ({
-        role: undefined,
-        styles: styles.loader,
-      }),
+      defaultProps: () =>
+        getA11Props('loader', {
+          role: undefined,
+          styles: resolvedStyles.loader,
+        }),
     })
   }
 
-  handleClick = (e: React.SyntheticEvent) => {
-    const { disabled } = this.props
-
+  const handleClick = (e: React.SyntheticEvent) => {
     if (disabled) {
       e.preventDefault()
       return
     }
 
-    _.invoke(this.props, 'onClick', e, this.props)
+    _.invoke(props, 'onClick', e, props)
   }
 
-  handleFocus = (e: React.SyntheticEvent) => {
-    _.invoke(this.props, 'onFocus', e, this.props)
+  const handleFocus = (e: React.SyntheticEvent) => {
+    _.invoke(props, 'onFocus', e, props)
   }
+
+  setEnd()
+
+  return (
+    <ElementType
+      {...rtlTextContainer.getAttributes({ forElements: [children] })}
+      {...getA11Props('root', {
+        onClick: handleClick,
+        disabled,
+        className: classes.root,
+        onFocus: handleFocus,
+        ...unhandledProps,
+      })}
+    >
+      {hasChildren && children}
+      {!hasChildren && loading && renderLoader()}
```

Comment:
I suggest a small refactor there to make it more readable and avoid useless functions:

```tsx
 const iconElement = Icon.create(icon, {
    defaultProps: () =>
      getA11Props('icon', {
        styles: resolvedStyles.icon,
        xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
      }),
  })

  return (
    <ElementType
      {...rtlTextContainer.getAttributes({ forElements: [children] })}
      {...getA11Props('root', {
        onClick: handleClick,
        disabled,
        className: classes.root,
        onFocus: handleFocus,
        ...unhandledProps,
      })}
    >
      {hasChildren ? (
        children
      ) : (
        <>
          {loading &&
            Loader.create(loader || {}, {
              defaultProps: () =>
                getA11Props('loader', {
                  role: undefined,
                  styles: resolvedStyles.loader,
                }),
            })}
          {iconPosition !== 'after' && iconElement}
          {Box.create(content, {
            defaultProps: () =>
              getA11Props('content', { as: 'span', styles: resolvedStyles.content }),
          })}
          {iconPosition === 'after' && iconElement}
        </>
      )}
    </ElementType>
  )
```